packages/@react-native-windows/cli/src/runWindows/utils/autolink.ts

```diff
@@ -591,6 +592,155 @@ async function updateAutoLink(
+    }
+  }
+  return changesNeeded;
+}
+
+function fixPackagesConfig(
+  useWinUI3: boolean,
+  project: WindowsProjectConfig,
+  checkMode: boolean,
+) {
+  const projectFile = path.join(
+    project.folder,
+    project.sourceDir,
+    project.project!.projectFile,
+  );
+  const packagesConfig = path.join(
+    path.dirname(projectFile),
+    'packages.config',
+  );
+
+  let changed = false;
+
+  if (fs.existsSync(packagesConfig)) {
+    // if we don't have a packages.config, then this is a C# project, in which case we use <PackageReference> and dynamically pick the right XAML package.
+    const packagesConfigContents = configUtils.readProjectFile(packagesConfig);
+    const packageElements = packagesConfigContents.documentElement.getElementsByTagName(
+      'package',
+    );
+    const winUIPropsPath = path.join(
+      resolveRnwRoot(project),
+      'PropertySheets/WinUI.props',
+    );
+    const winuiPropsContents = configUtils.readProjectFile(winUIPropsPath);
+    const winui2xVersion = configUtils.tryFindPropertyValue(
+      winuiPropsContents,
+      'WinUI2xVersion',
+    );
+    const winui3Version = configUtils.tryFindPropertyValue(
+      winuiPropsContents,
+      'WinUI3Version',
+    );
+
+    const dialects = [
+      {id: 'Microsoft.WinUI', version: winui3Version!},
+      {id: 'Microsoft.UI.Xaml', version: winui2xVersion!},
+    ];
+    const keepPkg = useWinUI3 ? dialects[0] : dialects[1];
+    const removePkg = useWinUI3 ? dialects[1] : dialects[0];
+
+    const nodesToRemove: Element[] = [];
+    let needsToAddKeepPkg = true;
+    for (let i = 0; i < packageElements.length; i++) {
+      const packageElement = packageElements.item(i)!;
+      const idAttr = packageElement!.getAttributeNode('id');
+      const id = idAttr!.value;
+      if (id === removePkg.id) {
+        nodesToRemove.push(packageElement);
+        changed = true;
+      } else if (id === keepPkg.id) {
+        changed =
+          changed || packageElement.getAttribute('version') !== keepPkg.version;
+        packageElement.setAttribute('version', keepPkg.version!);
+        needsToAddKeepPkg = false;
+      }
+    }
+
+    nodesToRemove.forEach(pkg =>
+      packagesConfigContents.documentElement.removeChild(pkg),
+    );
+
+    if (needsToAddKeepPkg) {
+      const newPkg = packagesConfigContents.createElement('package');
+
+      Object.entries(keepPkg).forEach(([attr, value]) => {
+        newPkg.setAttribute(attr, value as string);
+      });
+      newPkg.setAttribute('targetFramework', 'native');
+      packagesConfigContents.documentElement.appendChild(newPkg);
+      changed = true;
+    }
+
+    if (!checkMode && changed) {
+      const serializer = new XMLSerializer();
+      const output = serializer.serializeToString(packagesConfigContents);
+      const formattedXml = formatXml(output);
+      fs.writeFileSync(packagesConfig, formattedXml, {encoding: 'utf-8'});
+    }
+  }
+  return changed;
+}
+
+function formatXml(input: string) {
```

Comment:
Is there some existing "pretty xml" code we could call here rather than rolling our own?

Reply:
I tried using xml-formatter but it doesn't work with ES6 apparently. Filed https://github.com/chrisbottin/xml-formatter/issues/31 but I didn't want to block on it :)

Reply:
Please remove package from "package.json" if no longer being used.

That specific warning is actually more of an issue on our end not supporting certain libraries with ES6 default export.

Depending on usage, there is a syntactic fix, or a possible TypeScript compiler config change. I've been meaning to look into that one though.

Reply:
Yeah, I think that lib should work. Just make sure you do this:

`import * as formater from 'xml-formatter'`
instead of 
`import formatter from 'xml-formatter'`

Reply:
yeah I had done that but then I get an error when trying to run tests about some babel thing not supporting something else. I'll file a follow up for it

Reply:
@asklar the babel error is if you do `import x = require('x')`. Could you please try with `import * as x from 'x'` instead? There will be a different error if there is one.

Reply:
Worst case, you can just `require` the thing instead of import, with the consequence of losing typings.

Reply:
![image](https://user-images.githubusercontent.com/22989529/107321674-d1d49100-6a57-11eb-8707-d62da3333068.png)

Reply:
Thanks.

It looks like I misread the typings for the library as `module.exports = format` instead of `export = format`. The difference between those is that the former defines a JS interface (transpiled) that refers to CommonJS semantics. The latter defines a JS interface (transpiled) with ES6 semantics.

Would you be able to try `const formatter = require('xml-formatter').__esModule`?

The import will be non-type-safe. You can still import typings from the module though.

We can fix this when we have `esModuleInterop` repo wide. I've had that on my radar to look at.

Reply:
this builds but fails at runtime during tests:
![image](https://user-images.githubusercontent.com/22989529/107322937-3e508f80-6a5a-11eb-9d53-7332a9136adb.png)

Reply:
Hmm... Maybe waiting until esModuleInterop it is enabled repo wide would be the best move after all.

If you want to use it right now, you should still be able to call the function from the require. Might need to console.log the shape of the returned object to find the property you need.

Reply:
Oh wait, turns out you can just require it without anything special.

![image](https://user-images.githubusercontent.com/835219/107323818-cf743600-6a5b-11eb-95ae-fac814102ea9.png)

Reply:
thanks that seems to work